components/sync15/src/clients/engine.rs

```diff
@@ -31,10 +31,24 @@ struct Driver<'a> {
     command_processor: &'a dyn CommandProcessor,
     interruptee: &'a dyn Interruptee,
     config: &'a InfoConfiguration,
+    recent_clients: HashMap<String, RemoteClient>,
 }
 
 impl<'a> Driver<'a> {
-    fn sync(&self, inbound: IncomingChangeset) -> Result<OutgoingChangeset> {
+    fn new(
+        command_processor: &'a dyn CommandProcessor,
+        interruptee: &'a dyn Interruptee,
+        config: &'a InfoConfiguration,
+    ) -> Driver<'a> {
+        Driver {
+            command_processor,
+            interruptee,
+            config,
+            recent_clients: HashMap::new(),
+        }
+    }
+
+    fn sync(&mut self, inbound: IncomingChangeset) -> Result<OutgoingChangeset> {
```

Comment:
Are we sure that `inbound.changes` will contain all the `tabs` collection? Or will it only contain the clients that have changed since last sync?

Reply:
We fetch the entire `clients` collection on every sync, without a last modified timestamp (and never persist them anywhere), so yep, these will match! 👍

Reply:
I'll add a comment! 😁